### PR TITLE
Feature/issue136 bar modules

### PR DESCRIPTION
**Pull request by FelixSchladt**
I added a timestamp in the top right corner and changable colors

### CODE REVIEW

File: examples/get_time.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (janschaible) [inline examples/get_time.rs:1]: is this file still needed ??:)

File: tools/oxide-bar/src/config.rs [hunk 2]
```diff
@@ -4,25 +4,74 @@ use std::fs::File;
...
+                is_alpha: true,
+            }
+        } else {
+            Color {
+                red,
+                green,
+                blue,
+                alpha: 1.0,
+                is_alpha: false,
+            }
```
Comment (janschaible) [inline tools/oxide-bar/src/config.rs:36]: let alpha_present = hex.len()==9;
let alpha = if alpha_present {
            let alpha = u8::from_str_radix(&hex[7..9], 16).unwrap() as f64 / 255.0 as f64;
}
Color {
                red,
                green,
                blue,
                alpha,
                is_alpha: alpha_present,
}

?:)

File: tools/oxide-bar/src/main.rs [hunk 3]
```diff
@@ -197,45 +212,67 @@ impl OxideBar {
...
-        cr.show_text("Hi there").unwrap();
 
-        let ws_vec = state.get_workspace_list(self.screen);
+        //let (r, g, b) = self.config.color_txt.rgb();
+        //cr.set_source_rgb(r, g, b);
+        //let layout = create_layout(&cr).unwrap();
+        //println!("pixel size: {:?}", layout.pixel_size());
+        //layout.set_ellipsize(EllipsizeMode::End);
+        //layout.set_text("T");
+        //show_layout(&cr, &layout);
```
Comment (janschaible) [inline tools/oxide-bar/src/main.rs:242]: is this code still needed ?:)

File: tools/oxide-bar/src/main.rs [hunk 4]
```diff
@@ -276,7 +313,7 @@ fn get_x11rb_events(
     }
 }
 
-fn get_state(event_sender_mutex: Arc<Mutex<Sender<EventType>>>) {
+fn thread_state(event_sender_mutex: Arc<Mutex<Sender<EventType>>>) {
```
Comment (janschaible) [inline tools/oxide-bar/src/main.rs:316]: could you explain, why there is another thread started inside this function, which already runs in a seperate thread ?:)
couldn't you write a function state_signal_channel_async that blocks, until a state is received and then returns ?:)

File: tools/oxide-bar/src/main.rs (hunk 3 above)
Comment (FelixSchladt) [reply tools/oxide-bar/src/main.rs:242]: if this is  okay i would like to keep it. This would be the foundation for future modules. But I could extract this into an example

File: tools/oxide-bar/src/main.rs (hunk 4 above)
Comment (FelixSchladt) [reply tools/oxide-bar/src/main.rs:316]: This is an intermediate thread that awaits a state signal from the oxide-ipc and then transforms this into a oxide-bar datastructure. 
This is done in order to not need to implement oxide-bar specific datastructures into oxide ipc.
I think the performance hit is very minimal because each of the threads waits on the other one and therefore will need little to no runtime

**Review by janschaible — APPROVED** [review]
